src/main/java/com/icthh/xm/ms/otp/service/impl/OneTimePasswordServiceImpl.java

```diff
@@ -20,35 +24,75 @@
  * Service Implementation for managing OneTimePassword.
  */
 @Service
+@Slf4j
 @Transactional
 public class OneTimePasswordServiceImpl implements OneTimePasswordService {
 
-    private final Logger log = LoggerFactory.getLogger(OneTimePasswordServiceImpl.class);
+    private final OtpSpecService otpSpecService;
 
     private final OneTimePasswordRepository oneTimePasswordRepository;
 
     private final OneTimePasswordMapper oneTimePasswordMapper;
 
+    private final CommunicationService communicationService;
+
     public OneTimePasswordServiceImpl(OneTimePasswordRepository oneTimePasswordRepository,
-                                      OneTimePasswordMapper oneTimePasswordMapper) {
+                                      OneTimePasswordMapper oneTimePasswordMapper,
+                                      OtpSpecService otpSpecService,
+                                      CommunicationService communicationService) {
         this.oneTimePasswordRepository = oneTimePasswordRepository;
         this.oneTimePasswordMapper = oneTimePasswordMapper;
+        this.otpSpecService = otpSpecService;
+        this.communicationService = communicationService;
     }
 
     /**
      * Save a oneTimePassword.
      *
-     * @param oneTimePasswordDTO the entity to save
+     * @param oneTimePasswordDTO the entity to generate
      * @return the persisted entity
      */
     @Override
-    public OneTimePasswordDTO save(OneTimePasswordDTO oneTimePasswordDTO) {
-        log.debug("Request to save OneTimePassword : {}", oneTimePasswordDTO);
-        OneTimePassword oneTimePassword = oneTimePasswordMapper.toEntity(oneTimePasswordDTO);
-        oneTimePassword = oneTimePasswordRepository.save(oneTimePassword);
-        return oneTimePasswordMapper.toDto(oneTimePassword);
+    @Transactional
```

Comment:
`@Transactional` also for all class. Please, remove one of them as it is redundant.